### ISSUE TITLE
!empty check causes weird array type

### ISSUE THREAD

**Issue by weirdan** [vimeo/psalm#2333]
https://psalm.dev/r/4cbd39673e
```php
<?php
  /** @var array */
  $r = [];

  if (!empty($r['a'])) {}
  atan($r);
```
**Expected**: call to `atan` is reported with `array<array-key, mixed>` argument
**Actual**:
```
Psalm output (using commit 7b71991):

ERROR: InvalidArgument - 6:8 - Argument 1 of atan expects float, array{a: mixed}<array-key, mixed> provided

ERROR: UnusedFunctionCall - 6:3 - The call to atan is not used
```
Note the argument type: `array{a: mixed}<array-key, mixed>`

**Comment by muglug** [comment]
The argument is incorrect outside, however  that type would  be correct inside the condition (it's in  an unparseable form, but one that Psalm uses to represent complex array types internally).
